src/h3lib/include/geodesic_polygon.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2025 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @file geodesic_polygon.h
+ * @brief Internal data structures for geodesic polygon acceleration.
+ */
+
+#ifndef GEODESIC_POLYGON_INTERNAL_H
+#define GEODESIC_POLYGON_INTERNAL_H
+
+#include <stdbool.h>
+
+#include "bbox.h"
+#include "geodesic_cell_boundary.h"
+#include "polygon.h"
+#include "vec3d.h"
+
+/** @struct GeodesicEdge
+ *  @brief Cached data describing a polygon edge on the unit sphere.
+ */
+typedef struct {
+    Vec3d vert;       ///< Vertex position on the unit sphere
+    Vec3d edgeCross;  ///< Great-circle normal for the edge starting at vert
+    double edgeDot;   ///< Dot product between consecutive vertices
+    AABB aabb;        ///< Bounding box covering the edge arc
+} GeodesicEdge;
+
+/** @struct GeodesicLoop
+ *  @brief Polygon loop represented in geodesic space.
+ */
+typedef struct {
+    int numVerts;         ///< Number of vertices in the loop
+    GeodesicEdge *edges;  ///< Edge data for the loop
+    Vec3d centroid;       ///< Approximate centroid used for hemisphere tests
+} GeodesicLoop;
```

Comment:
if the length of `edges` is `numVerts`, seems like it would be easier to name it `verts` and `numVerts` or `edges` and `numEdges`?